File: app/src/main/java/com/example/consultants/match/ui/contactlist/ContactListViewModel.java
package com.example.consultants.match.ui.contactlist;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;
import android.support.annotation.NonNull;

import java.util.List;

public class ContactListViewModel extends AndroidViewModel {
    public static final String TAG = BookListViewModel.class.getSimpleName() + "_TAG";

    private BookRepository bookRepository;

    public BookListViewModel(@NonNull Application application) {
        super(application);
        bookRepository = new BookRepository(new RemoteDataSource(), new LocalDataSource(application.getApplicationContext()));
    }

    public LiveData<List<Book>> getBooks() {
        return bookRepository.getBooks();
    }

}

File: app/src/main/java/com/example/consultants/match/ui/contactlist/ContactListViewModel.kt
package com.example.consultants.match.ui.contactlist

import android.app.Application
import android.arch.lifecycle.AndroidViewModel
import android.arch.lifecycle.LiveData
import com.example.consultants.match.model.data.ContactRepository
import com.example.consultants.match.model.data.remote.RemoteDataSource
import com.example.consultants.match.model.jsondata.Contact

class ContactListViewModel(application: Application) : AndroidViewModel(application) {

    private val contactRepository: ContactRepository = ContactRepository(RemoteDataSource())

    val contacts: LiveData<List<Contact>>
        get() = contactRepository.getContacts()

}
